zenoh-plugin-ros2dds/src/route_service_cli.rs

```diff
@@ -373,32 +373,12 @@ fn route_dds_request_to_zenoh(
         is_little_endian,
     );
 
-    // Copy CDR Header
-    let cdr_header = match dds_req_buf.get(0..4) {
-        Some(hdr) => hdr,
-        None => {
-            // TODO: do we want this to return or explode ?
-            tracing::error!(
-                "Sample buffer was empty, Could not ger cdr header expected bytes [0..4]",
-            );
-            return;
-        }
-    };
-
+    // route request buffer stripped from request_id (client_id + sequence_number)
+    let mut zenoh_req_buf = ZBuf::empty();
+    // copy CDR Header
+    zenoh_req_buf.push_zslice(slice.subslice(0, 4).unwrap());
     // copy Request payload, skipping client_id + sequence_number
-    let payload = match dds_req_buf.get(20..dds_req_buf.len()) {
-        Some(hdr) => hdr,
-        None => {
-            // TODO: do we want this to return or explode ?
-            tracing::error!(
-                "Sample buffer payload subslice failed, expected bytes [20..buffer.len()]",
-            );
-            return;
-        }
-    };
-
-    let res: Vec<u8> = [cdr_header, payload].concat();
-    let zenoh_req_buf = ZBytes::new(res);
+    zenoh_req_buf.push_zslice(slice.subslice(20, slice.len()).unwrap());
```

Comment:
This gets rid of the unwraps, and doesnt add additional allocations. 

```rust
let mut zenoh_req_buf = ZBuf::empty();
match (slice.subslice(0, 4), slice.subslice(20, slice.len())) {
    (Some(cdr_header), Some(req_payload)) => {
        zenoh_req_buf.push_zslice(cdr_header);
        zenoh_req_buf.push_zslice(req_payload);
    }
    (None, Some(_)) => {
        tracing::warn!("Expected CDR Header at DDS_sample[0..4]");
        return;
    }
    (Some(_), None) => {
        tracing::warn!("Expected payload at DDS_sample[20..]");
        return;
    }
    (None, None) => {
        tracing::warn!("Could not get CDR header or DDS Payload");
        return;
    }
};
```

It is a little bit more verbose, but should fix the performance regression introduced from the vector allocation.

Reply:
Less Verbose Alternative
```rust
let mut zenoh_req_buf = ZBuf::empty();
match (slice.subslice(0, 4), slice.subslice(20, slice.len())) {
    (Some(cdr_header), Some(req_payload)) => {
        zenoh_req_buf.push_zslice(cdr_header);
        zenoh_req_buf.push_zslice(req_payload);
    }
    (_, _) => {
        tracing::warn!("{route_id} : received invalid request: {sample:0x?} -  Zenoh Payload Could not be created, Either CDRHeader or DDSRawSample Payload Could not be sliced into");
        return;
    }
};
```

Reply:
Addressed in latest commit. Please review!